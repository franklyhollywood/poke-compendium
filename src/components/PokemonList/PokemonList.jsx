import React from 'react';
import Pokemon from '../Pokemon/Pokemon.jsx';

export default function PokemonList({ pokemonList }) {
	console.log(pokemonList);
	return (
		<ul aria-label='list'>
			{/* <Pokemon /> */}
			{pokemonList.map(
				({ pokemon, _id, hp, ability_1, egg_group_1, pokebase, url_image }) => (
					<Pokemon
						pokemon={pokemon}
						hp={hp}
						ability_1={ability_1}
						egg_group_1={egg_group_1}
						pokebase={pokebase}
						url_image={url_image}
						key={_id}
					/>
				)
			)}
		</ul>
	);
}
