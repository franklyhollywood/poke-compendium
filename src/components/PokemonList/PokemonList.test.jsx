import { screen, render } from '@testing-library/react';
import PokemonList from './PokemonList.jsx';

it('testing my lists', () => {
	const fakePokes = [
		{
			pokemon: 'pikachu',
			hp: '60',
			ability_1: 'compound-eyes',
			egg_group_1: 'bug',
			pokebase: 'butterfree',
		},
		{
			pokemon: 'venusaur-mega',
			hp: '80',
			ability_1: 'compound-eyesthick-fat',
			egg_group_1: 'monster',
			pokebase: 'venusaur',
		},
	];
	render(<PokemonList pokemonList={fakePokes} />);
	const pokelist = screen.getByLabelText('list');
	expect(pokelist).toMatchSnapshot();
});
