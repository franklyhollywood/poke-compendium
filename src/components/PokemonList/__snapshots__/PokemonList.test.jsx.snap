// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing my lists 1`] = `
<ul
  aria-label="list"
>
  <div
    aria-label="pokemon"
  >
    <p>
      Name: 
      pikachu
    </p>
    <p>
      Health Points: 
      60
    </p>
    <p>
      Ability: 
      compound-eyes
    </p>
    <p>
      Egg Group: 
      bug
    </p>
    <p>
      Pokebase: 
      butterfree
    </p>
    <img
      alt="pikachu"
    />
  </div>
  <div
    aria-label="pokemon"
  >
    <p>
      Name: 
      venusaur-mega
    </p>
    <p>
      Health Points: 
      80
    </p>
    <p>
      Ability: 
      compound-eyesthick-fat
    </p>
    <p>
      Egg Group: 
      monster
    </p>
    <p>
      Pokebase: 
      venusaur
    </p>
    <img
      alt="venusaur-mega"
    />
  </div>
</ul>
`;
